src/Platforms.jl

```diff
@@ -110,9 +110,9 @@ const ARCHITECTURE_FLAGS = Dict(
         ),
         "armv7l" => OrderedDict(
             # Base armv7l architecture, with the most basic of FPU's
-            "armv7l"   => ["-march=armv7-a", "-mtune=generic-armv7-a", "-mfpu=vfpv3", "-mfloat-abi=hard"],
+            "armv7l"   => ["-mcpu=generic-armv7-a", "-mfpu=vfpv3", "-mfloat-abi=hard"],
```

Comment:
Looks like this broke armv7l: https://dev.azure.com/JuliaPackaging/BinaryBuilder.jl/_build/results?buildId=19048&view=logs&jobId=54759826-d195-5ca2-1571-21806663cd36&j=54759826-d195-5ca2-1571-21806663cd36&t=9eb882d6-5a34-5e37-1506-d0b4bf16214f
```
gfortran -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o -c hello_world.f
Assembler messages:
Error: unknown cpu `generic-armv7-a'
Error: unrecognized option -mcpu=generic-armv7-a
make: *** [Makefile:16: /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o] Error 1
Previous command exited with 2
gfortran linking specialty flags: Error During Test at /data/staticfloat/bb_azp/rootfs/agent_2/_work/1/s/test/building.jl:181
  Got exception outside of a @test
  Build for gfortran_flags on armv7l-linux-gnueabihf-libgfortran3 did not complete successfully
```
The weird thing is that I tested this locally before opening the PR:
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("armv7l", "linux"))'
sandbox:${WORKSPACE} # SUPER_VERBOSE=1 make -C /usr/share/testsuite
make: Entering directory '/usr/share/testsuite'
make[1]: Entering directory '/usr/share/testsuite/c/dyn_link'
cc -shared -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo/libfoo.so -fPIC -g -O2 -lm libfoo/libfoo.c
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0xc636d5da -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -shared -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo/libfoo.so -fPIC -g -O2 -lm libfoo/libfoo.c
cc -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/dyn_link -I/usr/share/testsuite/c/dyn_link/libfoo -fPIC -g -O2 -Wl,-z,origin -Wl,-rpath,$ORIGIN/libfoo -L/tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo -lfoo dyn_link.c
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x70fed1f7 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/dyn_link -I/usr/share/testsuite/c/dyn_link/libfoo -fPIC -g -O2 -Wl,-z,origin -Wl,-rpath,$ORIGIN/libfoo -L/tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo -lfoo dyn_link.c
make[1]: Leaving directory '/usr/share/testsuite/c/dyn_link'
make[1]: Entering directory '/usr/share/testsuite/c/hello_world'
cc -o /tmp/testsuite/arm-linux-gnueabihf/c/hello_world/hello_world -g -O2 hello_world.c
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0xe3f43cdd -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/c/hello_world/hello_world -g -O2 hello_world.c
make[1]: Leaving directory '/usr/share/testsuite/c/hello_world'
make[1]: Entering directory '/usr/share/testsuite/c/openmp'
gcc -o /tmp/testsuite/arm-linux-gnueabihf/c/openmp/openmp -g -O2 -fopenmp openmp.c
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0xfd2ee6cc -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/c/openmp/openmp -g -O2 -fopenmp openmp.c
make[1]: Leaving directory '/usr/share/testsuite/c/openmp'
make[1]: Entering directory '/usr/share/testsuite/cxx/dyn_link'
c++ -shared -o /tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/librepeater/librepeater.so -fPIC -g -O2 -lm librepeater/librepeater.cc
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++ -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x2989adb5 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -shared -o /tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/librepeater/librepeater.so -fPIC -g -O2 -lm librepeater/librepeater.cc
c++ -o /tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/dyn_link -I/usr/share/testsuite/cxx/dyn_link/librepeater -fPIC -g -O2 -Wl,-z,origin -Wl,-rpath,$ORIGIN/librepeater -L/tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/librepeater -lrepeater dyn_link.cc
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++ -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x17eb0e94 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/dyn_link -I/usr/share/testsuite/cxx/dyn_link/librepeater -fPIC -g -O2 -Wl,-z,origin -Wl,-rpath,$ORIGIN/librepeater -L/tmp/testsuite/arm-linux-gnueabihf/cxx/dyn_link/librepeater -lrepeater dyn_link.cc
make[1]: Leaving directory '/usr/share/testsuite/cxx/dyn_link'
make[1]: Entering directory '/usr/share/testsuite/cxx/hello_world'
c++ -o /tmp/testsuite/arm-linux-gnueabihf/cxx/hello_world/hello_world -g -O2 hello_world.cc
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++ -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x0849c0f9 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/cxx/hello_world/hello_world -g -O2 hello_world.cc
make[1]: Leaving directory '/usr/share/testsuite/cxx/hello_world'
make[1]: Entering directory '/usr/share/testsuite/fortran/hello_world'
gfortran -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o -c hello_world.f
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gfortran -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x3bf9fdf4 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o -c hello_world.f
cc -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world -g -O2 /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o -lgfortran -lm
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x55877e51 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world -g -O2 /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world.o -lgfortran -lm
gfortran -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world hello_world.f
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gfortran -D_GLIBCXX_USE_CXX11_ABI=0 -frandom-seed=0x36e22828 -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -o /tmp/testsuite/arm-linux-gnueabihf/fortran/hello_world/hello_world hello_world.f
make[1]: Leaving directory '/usr/share/testsuite/fortran/hello_world'
make: Leaving directory '/usr/share/testsuite'
sandbox:${WORKSPACE} # echo $?
0
```
````

Reply:
```console
sandbox:${WORKSPACE} # echo 'int main(){}' | /opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -x c - -mcpu=foo
arm-linux-gnueabihf-gcc: error: unrecognized argument in option ‘-mcpu=foo’
arm-linux-gnueabihf-gcc: note: valid arguments to ‘-mcpu=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
```
`generic-armv7-a` looks very much like a valid target.... :confused:

Reply:
I get the error with GCC 5.2.0, but not 4.8.5:
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("armv7l", "linux"); preferred_gcc_version=v"5")'
sandbox:${WORKSPACE} # SUPER_VERBOSE=1 make -C /usr/share/testsuite
make: Entering directory '/usr/share/testsuite'
make[1]: Entering directory '/usr/share/testsuite/c/dyn_link'
cc -shared -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo/libfoo.so -fPIC -g -O2 -lm libfoo/libfoo.c
/opt/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -D_GLIBCXX_USE_CXX11_ABI=1 -frandom-seed=0xc636d5da -L/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -Wl,-rpath-link,/opt/arm-linux-gnueabihf/arm-linux-gnueabihf/lib -mcpu=generic-armv7-a -mfpu=vfpv3 -mfloat-abi=hard -shared -o /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo/libfoo.so -fPIC -g -O2 -lm libfoo/libfoo.c
Assembler messages:
Error: unknown cpu `generic-armv7-a'
Error: unrecognized option -mcpu=generic-armv7-a
make[1]: *** [Makefile:14: /tmp/testsuite/arm-linux-gnueabihf/c/dyn_link/libfoo/libfoo.so] Error 1
make[1]: Leaving directory '/usr/share/testsuite/c/dyn_link'
make: *** [Makefile:45: build-project-c-dyn_link] Error 2
make: Leaving directory '/usr/share/testsuite'
```
Uhm....